notifier/main.py

```diff
@@ -102,22 +125,49 @@ async def _main() -> None:
                 "footer": f"{num_successful_checks}/{num_checks} checks have succeeded",
             }
 
-        response = await post_message(
-            channel=channel,
-            ts=message_ts,
-            icon_emoji=":package:",
-            attachments=[
-                {
-                    "text": status.message,
-                    "author_name": status.author.login,
-                    "author_icon": status.author.avatar_url,
-                    "footer": f"Commit: <{status.url}|`{status.sha[:7]}`>",
-                    "color": commit_color,  # type: ignore
-                },
-                *attachments,
-                summary,
-            ],
-        )
+        response = None
+
+        # Attempt to post notification. If post fails retry provided number of times.
+        for attempt in range(1, notify_retries + 1):
+            try:
+                response = await post_message(
+                    channel=channel,
+                    ts=message_ts,
+                    icon_emoji=":package:",
+                    attachments=[
+                        {
+                            "text": status.message,
+                            "author_name": status.author.login,
+                            "author_icon": status.author.avatar_url,
+                            "footer": f"Commit: <{status.url}|`{status.sha[:7]}`>",
+                            "color": commit_color,  # type: ignore
+                        },
+                        *attachments,
+                        summary,
+                    ],
+                    timeout=notify_timeout,
+                )
+                break
+            except (
+                httpx.ConnectTimeout,
+                httpx.PoolTimeout,
+                httpx.ReadTimeout,
+                httpx.WriteTimeout,
+            ):
+                if attempt == notify_retries:
+                    logger.error(
+                        "%s attempt failed. Unable to post message to slack channel",
+                        attempt,
+                    )
+                    raise
+
+                sleep_duration = 0.5 * attempt
+                logger.error(
+                    "Unable to post message to slack channel (attempt %s), retrying in %s seconds",
```

Comment:
```suggestion
                    "Timed out while trying to post message to slack channel (attempt %s), retrying in %s seconds",
```
Since we're just catching timeout exceptions :)